server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3883,6 +3899,15 @@ public boolean isRemoteTranslogEnabled() {
         return indexSettings() != null && indexSettings().isRemoteTranslogStoreEnabled();
     }
 
+    /**
+     * This checks if we are in state to upload to remote store. Until the cluster-manager informs the shard through
+     * cluster state, the shard will not be in STARTED state. This method is used to prevent pre-emptive segment or
+     * translog uploads.
+     */
+    public boolean isStartedPrimary() {
+        return getReplicationTracker().isPrimaryMode() && state() == IndexShardState.STARTED;
+    }
```

Comment:
will this affect the shards we are initializing from snapshot/remote store restore/local recovery ? As per the code , it shouldn't  . But would be good to double check this .

Reply:
have planned this already - https://github.com/opensearch-project/OpenSearch/pull/11330/#issuecomment-1826215993